src/source/statsd.rs

```diff
@@ -164,20 +166,25 @@ impl source::Source<StatsdConfig> for Statsd {
     }
 
     fn run(
-        mut self,
+        self,
         mut chans: util::Channel,
         tags: &sync::Arc<metric::TagMap>,
         poller: mio::Poll,
     ) -> () {
         for (idx, socket) in self.conns.iter() {
-            poller
+            match poller
                 .register(
                     socket,
                     mio::Token::from(idx),
                     mio::Ready::readable(),
                     mio::PollOpt::edge(),
                 )
-                .unwrap();
+            {
+                Ok(_) => {}
```

Comment:
Since we're not doing anything with the Ok arm, you can probably just use `if let Err(e) == poller.register... { error!....}`

Reply:
Fixed.

Reply:
Endorse.